specifyweb/frontend/js_src/lib/components/FormPlugins/LatLongUi.tsx

```diff
@@ -97,6 +103,15 @@ function Coordinate({
 
     isChanging.current = true;
 
+    const latLongBlockers = isValid
+      ? []
+      : [
+          fieldType === 'Lat'
+            ? localityText.validLatitude()
+            : localityText.validLongitude(),
+        ];
+    setBlockers(latLongBlockers, fieldType);
```

Comment:
on line 34 you have a `setValidation` callback, that should also set validation blockers - can you check if you can reuse that?
and I think the call to setBlockers/setValidation should be in useEffect so that it's not made on each render

Reply:
though, `setValidation` is already called here so validation should be working:
https://github.com/specify/specify7/blob/4e5722aabb4ae3e8afd2ee487f4bea68eaff83d1/specifyweb/frontend/js_src/lib/components/FormPlugins/LatLongUi.tsx#L88-L89

instead of adding extra validation code, could you please check why the existing validation code is not working?

Reply:
Looks like `setValidation` gets called twice somehow and so validation is not working as expected. 
This line should be blocking save correctly:
https://github.com/specify/specify7/blob/4e5722aabb4ae3e8afd2ee487f4bea68eaff83d1/specifyweb/frontend/js_src/lib/components/FormPlugins/LatLongUi.tsx#L89
But `setValidation` gets called once again due to this useEffect with an empty message which unsets the save blocker. I'm guessing the useEffect gets triggered due to some re-render in `useResourceValue` but still need to figure out why.
https://github.com/specify/specify7/blob/f4fc143eb5b31e1865d31e26533aa5297b3930bc/specifyweb/frontend/js_src/lib/hooks/useValidation.tsx#L116

Reply:
I think a straight-forward but maybe not ideal solution would be to make `useResourceValue` return `setBlockers` and use that in `LatLongUi` instead of `setValidation`. What do you think?

Reply:
yes, it seems like the setBlockers() call in useResourceValue changes the blockers array, which are passed to useValidation and in turn causing it to re-set (in this case empty) validation messages.
A fix would be to integrate the useValidation and useSaveBlockers closer together to prevent them from stepping on each other
Since setBlockers will indirectly call setValidation, making useResourceValue return setBlockers seems like a good idea - give it a try.

Edge cases to consider:
- there may be timing differences between when error is shown on the screen by setValidation vs setBlocker, and how the error is shown to the user (it is shown right away, or only only if field looses/gains focus?)
- setValidation accepts a key parameter (added by Jason in https://github.com/specify/specify7/commit/4879cd59fb28dbc63d7864bafc1ea3649c740b21). I think that instead of returning setBlockers directly, it might be better for useResourceValue to return something like `React.useCallback((blockers:provideTypingHere)=>setBlockers(blockers, 'customValidation'),[setBlockers]);` (could replace `customValidation` with any other string or variable) - point being that in places where useResourceValue is used, you won't be inconvenienced by having to come up with a key.

---

additional context:

The purpose of the useValidation hook (used by useResourceValue behind the scenes to get setValidation) is to set validation messages on some html input.
If there is validation error in an HTML input, the submit button will get disabled (by browser), and so saving would be prevented until errors are fixed (with great a11y benefits and screen reader support in the process)

Reply:
I'm choosing to return `setBlockers` directly as `LatLongUi.tsx` would need to specify a different key for longitude and latitude. I could add an optional key param to your callback but I'm not sure we're gaining anything from that